File: src/php/App/ValueObjects/StationAggregate.php
<?php

namespace App\ValueObjects;

use Stringable;

class StationAggregate implements Stringable
{
    public string $station;
    public int $itemCount;
    public float $temperatureMinimum;
    public float $temperatureMaximum;
    public float $temperatureSum;

    public function __construct(Detection $detection)
    {
        $this->station = $detection->station;
        $this->itemCount = 1;
        $this->temperatureMinimum = $detection->temperature;
        $this->temperatureMaximum = $detection->temperature;
        $this->temperatureSum = $detection->temperature;
    }

    public function addDetection(Detection $detection): static
    {
        $this->temperatureMinimum = min($this->temperatureMinimum, $detection->temperature);
        $this->temperatureMaximum = max($this->temperatureMaximum, $detection->temperature);
        $this->itemCount++;
        $this->temperatureSum += $detection->temperature;

        return $this;
    }

    private function temperatureMean(): float
    {
        return $this->temperatureSum / $this->itemCount;
    }

    public function __toString(): string
    {
        $roundedMean = round($this->temperatureMean(),1);

        return sprintf(
            '%s=%.1f/%.1f/%.1f',
            $this->station, $this->temperatureMinimum, $roundedMean, $this->temperatureMaximum
        );
    }
}

File: src/go-seq/internal/value_objects/station_aggregate.go
package value_objects

import (
	"fmt"
	"math"
)

type StationAggregate struct {
	Station     string
	ItemCount   uint64
	Temperature struct {
		Minimum, Maximum, Sum float64
	}
}

func NewStationAggregateFromDetection(d Detection) (sa StationAggregate) {
	sa.Station = d.Station
	sa.ItemCount = 1
	sa.Temperature.Sum = float64(d.Temperature)
	sa.Temperature.Minimum = float64(d.Temperature)
	sa.Temperature.Maximum = float64(d.Temperature)

	return
}

func (sa *StationAggregate) AddDetection(d Detection) {
	t := float64(d.Temperature)

	sa.ItemCount++
	sa.Temperature.Sum += t
	sa.Temperature.Minimum = min(sa.Temperature.Minimum, t)
	sa.Temperature.Maximum = max(sa.Temperature.Minimum, t)
}

func (sa *StationAggregate) TemperatureMean() float64 {
	return sa.Temperature.Sum / float64(sa.ItemCount)
}

func (sa *StationAggregate) String() string {
	roundedMean := math.Round(sa.TemperatureMean()*10) / 10
	return fmt.Sprintf(
		"%s=%.1f/%.1f/%.1f",
		sa.Station, sa.Temperature.Minimum, roundedMean, sa.Temperature.Maximum,
	)
}
